src/js/game/hud/parts/building_placer_logic.js

```diff
@@ -114,6 +120,34 @@ export class HUDBuildingPlacerLogic extends BaseHUDPart {
         this.root.camera.upPostHandler.add(this.onMouseUp, this);
     }
 
+    /**
+     * Returns the current base rotation for the current meta-building.
+     * @returns {number}
+     */
+    get currentBaseRotation() {
+        const rotationByBuilding = this.root.app.settings.getAllSettings().rotationByBuilding;
+        if (!rotationByBuilding) {
+            return this.currentBaseRotationGeneral;
+        }
+        const id = this.currentMetaBuilding.get().getId();
+        return this.preferredBaseRotations[id] == null
```

Comment:
If `this.preferredBaseRotations[id]` is `0`, comparing it with `null` will be truthy.

Instead do if `typeof this.preferredBaseRotations[id] === 'undefined'` or even better: this.preferredBaseRotations.hasOwnProperty(id).

Since you are using a map here though, actually the *best* solution would be simply to use a `new Map()`

Reply:
I ended up going for the hasOwnProperty() option.